### PR TITLE
Matrix operations

### PR DESCRIPTION
**Pull request by tudny**

### CODE REVIEW

File: src/matrices.rs [hunk 1]
```diff
@@ -17,8 +22,191 @@ impl<T: MatrixNumber> Matrix<T> {
...
+
+        Ok(Aftermath {
+            result: self,
+            steps: steps,
+        })
+    }
+
+    fn check_shape(&self, other: &Self) -> anyhow::Result<()> {
+        if self.data.len() != other.data.len() {
+            return Err(anyhow::anyhow!("Matrices have different number of rows!"));
```
Comment (mgr0dzicki) [inline src/matrices.rs]: `return Err(anyhow::anyhow!(...));` $\equiv$ `anyhow::bail!(...)`

**Comment by tudny** [comment]
This was the ugliest conflict I had to deal with

File: src/matrices.rs [hunk 2]
```diff
@@ -40,24 +248,280 @@ impl<T: MatrixNumber> LaTeXable for Matrix<T> {
...
+#[macro_export]
+macro_rules! im {
+    ($($($x:expr),+ $(,)?);+ $(;)?) => (
+        Matrix::new(vec![
+            $(vec![
+                $($x),+
+            ]),+
+        ])
+    );
+}
```
Comment (tudny) [inline src/matrices.rs]: @mgr0dzicki check out those wild macros!

File: src/main.rs [hunk 3]
```diff
@@ -1,3 +1,5 @@
+extern crate core;
```
Comment (mgr0dzicki) [inline src/main.rs]: why

**Review by mgr0dzicki — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 3 above)
Comment (tudny) [reply src/main.rs]: who asked

**Review by mgr0dzicki — APPROVED** [review]

**Review by mgr0dzicki — APPROVED** [review]

**Review by mgr0dzicki — APPROVED** [review]
niech ci będzie

**Pull request merged**